tutorial/source/dirichlet_process_mixture.ipynb

```diff
@@ -0,0 +1,790 @@
+        "\n",
+        "Furthermore, we can easily extend this to define the generative process of a nonparametric mixture model: every table $t$ that has at least one customer seated is associated with a set of cluster parameters $\\theta_t$, which were themselves drawn from some base distribution $G_0$. For each new observation, first assign that observation to a table according to the above probabilities; then, that observation is drawn from the distribution parameterized by the cluster parameters for that table. If the observation was assigned to a new table, draw a new set of cluster parameters from $G_0$, and then draw the observation from the distribution parameterized by those cluster parameters.\n",
+        "\n",
+        "While this formulation of a Dirichlet process mixture model is intuitive, it is also very difficult to perform inference on in a probabilistic programming framework. This motivates an alternative formulation of DPMMs, which has empirically been shown to be more conducive to inference (e.g. Blei and Jordan, 2004).\n",
+        "\n",
+        "## The Stick-Breaking Method (Sethuraman, 1994)\n",
+        "\n",
+        "The generative process for the stick-breaking formulation of DPMMs proceeds as follows:\n",
+        "\n",
+        "* Draw $\\beta_i \\sim \\text{Beta}(1, \\alpha)$ for $i \\in \\mathbb{N}$\n",
+        "* Draw $\\theta_i \\sim G_0$ for $i \\in \\mathbb{N}$\n",
+        "* Construct the mixture weights $\\pi$ by taking $\\pi_i(\\beta_{1:\\infty}) = \\beta_i \\prod_{j<i} (1-\\beta_j)$\n",
+        "\n",
+        "* For each observation $n \\in \\{1, ..., N\\}$, draw $z_n \\sim \\pi(\\beta_{1:\\infty})$, and then draw $x_n \\sim f(\\theta_{z_n})$\n",
+        "\n",
+        "Here, the infinite nature of the Dirichlet process mixture model can more easily be seen. Furthermore, all $\\beta_i$ are independent, so it is far easier to perform inference in a probabilistic programming framework.\n",
+        "\n",
+        "First, we import all the modules we're going to need:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "VaAgD92IdE6i",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "import torch\n",
+        "import torch.nn.functional as F\n",
+        "import pandas as pd\n",
+        "import numpy as np\n",
+        "import matplotlib.pyplot as plt\n",
+        "import matplotlib.cm as cm\n",
+        "from tqdm import tqdm\n",
+        "\n",
+        "from pyro.distributions import *\n",
+        "import pyro\n",
+        "from pyro.optim import Adam\n",
+        "from pyro.infer import SVI, Trace_ELBO\n",
+        "from torch.distributions import constraints"
```

Comment:
Could you enable validation and set random seed here?
```
assert pyro.__version__.startswith('1.0.0')
pyro.enable_validation(True)       # can help with debugging
pyro.set_rng_seed(0)
```
It would be clearer if you move `from torch.distributions import constraints` to the above import block.